src/lib/model.ts

```diff
@@ -216,6 +217,18 @@ export class Course {
 		return this.modules
 	}
 
+	isOptional() {
```

Comment:
Would `containsOnlyOptionalModules` be a more appropriate name for this? Or can it be called an "optional course"?